app/__init__.py

```diff
@@ -97,12 +111,24 @@ def plan():
     return redirect(url_for("plan_error"))
 
 
-@app.route("/plan/<id>")
+@app.route("/plan/<id>", methods=["GET"])
 def plan_from_db(id):
     plan = Plan.query.get_or_404(id, description="This plan doesn't exist.")
+    if plan.password_hash:
+        return render_template("locked.html")
     return render_template("plan.html", plan=plan)
 
 
+@app.route("/plan/<id>", methods=["POST"])
+def plan_from_db_with_password(id):
+    plan = Plan.query.get_or_404(id, description="This plan doesn't exist.")
+    form = PasswordForm()
+    form.validate_on_submit()
+    if check_password_hash(plan.password_hash, form.password.data):
```

Comment:
What would happen if a user POST a password for an unprotected plan ? Is it equivalent to this ?

``` python
 security.check_password_hash(None, "myhash")
 ```